elsie/inkscape.py

```diff
@@ -0,0 +1,118 @@
+import contextlib
+import logging
+import os
+import queue
+import subprocess
+import tempfile
+import threading
+
+PROMPT = "> "
+
+
+class ProcessReader:
+    def __init__(self, stdout, queue):
+        self.thread = threading.Thread(
+            target=self.read, args=(stdout, queue), daemon=True
+        )
+
+    def start(self):
+        self.thread.start()
+
+    def read(self, stdout, queue):
+        line = ""
+
+        while True:
+            character = stdout.read(1)
+            line += character
+            if character == "\n" or line == PROMPT:
+                queue.put(line)
+                line = ""
+
+
+@contextlib.contextmanager
+def svg_file_input(inkscape, svg: str, binary=False):
+    with tempfile.NamedTemporaryFile("wb" if binary else "w", suffix=".svg") as f:
+        f.write(svg)
```

Comment:
Probably will not work on Windows due to file writer lock.

Reply:
We can solved it together with the first windows user :)

Reply:
Do you need something tested under windows?  I have inkscape and Py 3.7 installed, wouldn't be too tough to try some examples.

Reply:
To be honest, we haven't really tested `elsie` under Windows at all. You can try the `master` branch if you can compile some of the examples in the repository root.